Updated confirm text for switching catalog. Updated titles and help text

dmundra opened the pull request:
Closes https://github.com/GSA/openacr/issues/340 and fixes https://github.com/GSA/openacr/issues/341

Summary of changes:
* Updated select catalog text.
* Moved confirmation to a button.
* Added reset button to go with the confirm button.
* Update tests to use the confirm button.
* Added tests for the checking the confirm and reset button dialogs and functionality.

**QA Help Text**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. On the about page confirm you see 'Select catalog' accordion.
4. Confirm the accordion is expanded and has a radio choices below the text 'Select which catalog you will be using for the OpenACR.'
5. Confirm the choice text and help text.
6. Switch the choice and click the 'Confirm' button and then confirm the dialog text.

**QA cancel change**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. On the about page confirm you see 'Select catalog' accordion.
4. Switch the choice and click the 'Confirm' button and then cancel the dialog text.
5. Switch to a chapter page or report page and confirm the catalog has not changed.
6. Switch back to the 'about' page and confirm the choice is reset to the existing one.

**QA new report**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. Switch the choice from WCAG 2.0 to any of the other choices.
4. Click the 'Confirm' button and confirm the change.
5. Confirm the chapters in the navigation changes and the sidebar changes when switching the choices.
7. Selecting WCAG 2.1 catalog.
8. Click the 'Confirm' button and confirm the change.
9. Navigate to the chapter pages and confirm you see WCAG 2.1 criteria and links go to WCAG 2.1 pages.
10. Add some text, select some values.
11. Click the 'View report' or 'Report' tab.
12. Confirm you see a green 'Valid Report' box at the top of the report page with the message 'Your report has passed validation'.
13. Confirm the report has the WCAG 2.1 criteria and links go to WCAG 2.1 pages.
14. Download the YAML & HTML report.
15. Confirm the report has the WCAG 2.1 criteria and links go to WCAG 2.1 pages.
16. Go back to the 'Overview' page
17. Click 'Start new report' or 'New report'.
18. Click 'Open report'.
19. Upload the downloaded YAML file from step 13.
20. Confirm the catalog selection shows the WCAG 2.1 selection.

**QA existing report**

1. After deployment ('Deploy' action) is successful, open site https://gsa.github.io/openacr-editor/.
2. Click 'Start new report' or 'New report'.
3. Click 'Open report'.
4. Upload a yaml file from OpenACR that already has 'last modified date' and 'version' set to some values (if needed modify the YAML file to set those values before uploading). Try the 'drupal-9.yaml' or any of the yaml files from https://github.com/GSA/openacr/tree/main/openacr where last modified date is 11/16/2021 and version is 12.
5. Confirm the catalog selection shows the WCAG 2.0 selection.
7. Switch the choice from WCAG 2.0 to any of the other choices.
8. Click the 'Confirm' button and confirm the change.
9. Navigate to the chapter pages and confirm you see WCAG 2.1 criteria and links go to WCAG 2.1 pages.
10. Add some text, select some values.
11. Click the 'View report' or 'Report' tab.
12. Confirm you see a green 'Valid Report' box at the top of the report page with the message 'Your report has passed validation'.
13. Confirm the report has the WCAG 2.1 criteria and links go to WCAG 2.1 pages.
14. Download the YAML & HTML report.
15. Confirm the report has the WCAG 2.1 criteria and links go to WCAG 2.1 pages.
16. Go back to the 'Overview' page
17. Click 'Start new report' or 'New report'.
18. Upload the downloaded YAML file from step 13.
19. Confirm the catalog selection shows the WCAG 2.1 selection.
20. Switch the choice back to WCAG 2.0.
21. Click the 'Confirm' button and confirm the change.
22. Navigate to the chapter pages and confirm you only see WCAG 2.0 criteria and links go to WCAG 2.0 pages.
23. Click the 'View report' or 'Report' tab.
24. Confirm you see a green 'Valid Report' box at the top of the report page with the message 'Your report has passed validation'.
25. Confirm the report has the only see WCAG 2.0 criteria and links go to WCAG 2.0 pages.

danid123 reviewed (changes requested):
@dmundra When the user hits cancel on the warning it does the same thing as ok. Is it possible to make cancel not change the radio so that the user can save?

I would also add this line "Select cancel to download your report from the Report page before proceeding." So the full text would be:

> Switching catalogs may remove entered data and notes from your ACR that are not part of the newly selected catalog. Please download your report before switching catalogs to avoid losing information. Select cancel to download your report from the Report page before proceeding.

Can we add "in the United States of America." to the end of the first option. It would make the three options parallel:
> These are the minimum standards required for Section 508 conformance in the United States of America.

Thanks!

dmundra commented:
@danid123 updated the text and created the follow-up bug: https://github.com/GSA/openacr/issues/341

dmundra commented:
@danid123 and @mgifford I fixed the bug with the radio button not ignore the selection when you cancel the confirmation. 

To fix the bug I needed to add a button that will confirm the choice. This is the first time we added a button to "Save" a change so please review and let me know if the text needs to be updated.

mgifford commented:
Can we default to the Current best practice? VPAT® 2.4 508 + WCAG: Revised Section 508 Edition (WCAG 2.1)

Defaulting to the minimum sets those expectations. Folks should have to choose a lower standard if they don't want to follow the best practice?

mgifford commented:
Data loss really should only occur if you are going from 
- 508 or 508 + WCAG  --> WCAG
- 508 + WCAG  --> 508 

Should we warn folks if they are going from:
- WCAG  --> 508 + WCAG
- 508 --> 508 + WCAG

I think I'm missing one...

Maybe we could congratulate them, but warn them that they might loose data if they go back? Not sure. That's probably an enhancement.

mgifford commented:
The catalog type should be appended to the file name. 
- 508
- WCAG
- WCAG+508

Probably an enhancement

dmundra commented:
@mgifford thank you. Text and default changes makes sense so I will make them here.

I might be able to add the data loss text only when you are going from 2.1 to 2.0 but ya an enhancement ticket might be better.

As for adding the catalog to the filename, definitely separate ticket. To note we do add the catalog type to the YAML file.

dmundra commented:
@mgifford changed the default to VPAT® 2.4 508 + WCAG: Revised Section 508 Edition (WCAG 2.1) and updated the tests for that. Also added the line breaks to the dialog text. Created tickets for the enhancements.

danid123 commented:
- [ ] Instead of: 
> Select which catalog you want to use for this OpenACR and confirm the change. 
> Report type change is not saved till you confirm.
Can we change it to: 
> Select which catalog you want to use for this OpenACR.
> Confirm the new report type to save your new selection.
- [ ] Change the button styles 
![Buttons](https://user-images.githubusercontent.com/4855793/156982462-02fbf19a-d0f4-4ae5-a36b-ed10a390771e.png)
and label to: 
- [ ] "OpenACR passed validation when switching catalog." Is there a more plain language way of saying this? @mgifford any ideas? It's a little too machine sounding.

- [ ] "Are you sure this is what you would like to do?" Is confusing in this context since there are two buttons that do different things. It would make the user think that canceling is destructive when it's not. I would just remove it. Or it should be the headline.
Ideally the popup would look like this (format and cancel pattern from USWDS):
![Chaning catalogs popup - 1](https://user-images.githubusercontent.com/4855793/156980151-29bdc416-9e6e-4081-bedf-2c95c5fb8322.png)
Text: Switching catalogs may remove entered data and notes from your ACR that are not part of the newly selected catalog
Please download your report before switching catalogs to avoid losing information. Select Go Back to save first.

But if that's too much, I would just do this:
![Chaning catalogs popup - 2](https://user-images.githubusercontent.com/4855793/156982211-6e4e66d1-f306-471b-bd0f-fd00a8a45890.png)
Text: Switching catalogs may remove entered data and notes from your ACR that are not part of the newly selected catalog
Please download your report before switching catalogs to avoid losing information. Select Cancel to save before switching.

danid123 commented:
I think the disabled buttons look a little odd with the strikethrough but they do have higher contrast than the USWDS disabled buttons.

dmundra commented:
@danid123 thank you for you feedback. I made some of the changes and here my responses to the others:

> "OpenACR passed validation when switching catalog." Is there a more plain language way of saying this? @mgifford any ideas? It's a little too machine sounding.

It is a great point. I mostly had it in there when I discovered the validator didn't like having extra criteria when comparing against the different catalog. We can just change to a simple success message if that is preferred.

> Switching to USWDS dialog

Right now I am using the browser [built-in confirm dialog](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm) and there it no options to tweak it. A custom dialog based on USWDS or a component would be a lot more work right now so I think it would be an issue in the backlog. Can you create that issue for the future?

danid123 commented:
I created [#345](https://github.com/GSA/openacr/issues/345) to address switching to the USWDS component style and UI pattern
